src/ChunkMap.cpp

```diff
@@ -1088,7 +1088,7 @@ int  cChunkMap::GetHeight(int a_BlockX, int a_BlockZ)
 		}
 
 		// The chunk is not valid, wait for it to become valid:
-		cCSUnlock Unlock(Lock);
+		m_CSLayers.Unlock();
 		m_evtChunkValid.Wait();
```

Comment:
I've changed the code above to lock the mutex, rather than using cCSLock, so this code should now be fixed.